### PR TITLE
remove v6.5.x and v6.6.x version constants

### PR DESCRIPTION
**Pull request by talevy**
related to refactoring initiative #41164.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/AutoFollowStats.java [hunk 1]
```diff
@@ -116,14 +114,9 @@ public AutoFollowStats(StreamInput in) throws IOException {
...
-        recentAutoFollowErrors = new TreeMap<>(in.readMap((Writeable.Reader<String>) StreamInput::readString,
-                (Writeable.Reader<Tuple<Long, ElasticsearchException>>) in1 -> new Tuple<>(in1.readZLong(), in1.readException())));
-        if (in.getVersion().onOrAfter(Version.V_6_6_0)) {
-            autoFollowedClusters = new TreeMap<>(in.readMap(StreamInput::readString, AutoFollowedCluster::new));
-        } else {
-            autoFollowedClusters = Collections.emptyNavigableMap();
-        }
+        recentAutoFollowErrors = new TreeMap<>(in.readMap(StreamInput::readString,
+            in1 -> new Tuple<>(in1.readZLong(), in1.readException())));
+        autoFollowedClusters = new TreeMap<>(in.readMap(StreamInput::readString, AutoFollowedCluster::new));
```
Comment (jpountz) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/AutoFollowStats.java]: can you retain the comment and the cast?

**Review by jpountz — APPROVED** [review]
One minor comment, other than that LGTM.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/AutoFollowStats.java (hunk 1 above)
Comment (talevy) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/AutoFollowStats.java]: thanks for the catch! I must have deleted the comment initially, and then forgot why the cast was there, and then removed the cast. I've returned both.

**Comment by talevy** [comment]
run elasticsearch-ci/1

**Pull request merged**